pkg/oc/cli/cmd/create/imagestreamtag.go

```diff
@@ -0,0 +1,208 @@
+package create
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	ocmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+const ImageStreamTagRecommendedName = "imagestreamtag"
+
+var (
+	imageStreamTagLong = templates.LongDesc(`
+		Create a new image stream tag
+
+		Image streams tags allow you to track, tag, and import images from other registries. They also 
+		define an access controlled destination that you can push images to. An image stream tag can 
+		reference images from many different registries and control how those images are referenced by 
+		pods, deployments, and builds.
+
+		If --resolve-local is passed, the image stream will be used as the source when pods reference
+		it by name. For example, if stream 'mysql' resolves local names, a pod that points to
+		'mysql:latest' will use the image the image stream points to under the "latest" tag.`)
+
+	imageStreamTagExample = templates.Examples(`
+		# Create a new image stream tag based on an image on a remote registry
+		%[1]s mysql:latest --from-image=myregistry.local/mysql/mysql:5.0
+		`)
+)
+
+type CreateImageStreamTagOptions struct {
+	ISTag  *imageapi.ImageStreamTag
+	Client client.ImageStreamTagsNamespacer
+
+	FromImage   string
+	From        string
+	Annotations []string
+
+	DryRun bool
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateImageStreamTag is a command to create a new image stream tag.
+func NewCmdCreateImageStreamTag(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateImageStreamTagOptions{
+		Out: out,
+		ISTag: &imageapi.ImageStreamTag{
+			ObjectMeta: metav1.ObjectMeta{},
+			Tag:        &imageapi.TagReference{},
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME",
+		Short:   "Create a new image stream tag.",
+		Long:    imageStreamTagLong,
+		Example: fmt.Sprintf(imageStreamTagExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"istag"},
+	}
+
+	cmd.Flags().StringVar(&o.FromImage, "from-image", "", "Use the provided remote image with this tag.")
+	cmd.Flags().StringVar(&o.From, "from", "", "Use the provided image stream tag or image stream image as the source: [<namespace>/]name[:<tag>|@<id>]")
```

Comment:
This is confusing to me and even more for users. We're currently using both for IS and docker image, see [here](https://github.com/openshift/origin/blob/2af10a7460093c1c4b74582622bee4eee1854f9e/pkg/oc/cli/cmd/importimage.go#L62) and [here](https://github.com/openshift/origin/blob/2af10a7460093c1c4b74582622bee4eee1854f9e/pkg/oc/cli/cmd/set/triggers.go#L163) not sure where else. We need to standardize on the naming and use it consistently everywhere.

Reply:
Yeah, I think having `source` and `from` is worse than `from` and `from-image` and `from-X`.  After all, that's consistent with `oc create` in general for `oc create secrets` which has `from-file` and `from-literal`.

I think in general source and from are first gen, while `from` and `from-X` are gen2

Reply:
What I was hoping is that we unify this across all the places, so that user whenever see flag `from` or `from-image` or else knows what can expect. Either this can be solved as part of this PR or an issue, please.

Reply:
But we can't break backwards compatibility, so it's more of a long running thing.  I'd say a card is more appropriate, but we're not going to fix it anytime soon (i.e. i don't want to change tag or import-image right now, i'd want to ask "are we going to write an improved import-image that handles it")

Reply:
Card sgtm.

Reply:
I agree and I am aware of backwards compatibility :/ But at the same time it hurts our UX.